### PR TITLE
Fix `InputWithDecorations` JSDocs

### PR DESCRIPTION
**Pull request by rohan-kadkol**
## Changes

`InputWithDecorations`'s JSDocs refers a non-existent `InputWithDecorations.Icon` component. This PR replaces that with `Icon`.

I also noticed that the example in the JSDocs for InputWithDecorations was not formatted correctly. This PR also fixes this by replacing `@usage` with `@example`.

|Before|After|
|-|-|
|image|image|

## Testing

* Confirmed that the example in `InputWithDecorations`'s JSDocs is now formatted correctly.
* Did a global search for `@usage` and found no other such issues.
* Did a global search for `InputWithDecorations.Icon` and confirmed that there are no more references to this non-existent component.

## Docs

N/A

### CODE REVIEW

**Review by AdamMeza-Bentley — APPROVED** [review]

File: packages/itwinui-react/src/core/InputWithDecorations/InputWithDecorations.tsx [hunk 1]
```diff
@@ -81,7 +81,7 @@ const InputWithDecorationsButton = React.forwardRef((props, ref) => {
  *
  * If you are not using default InputWithDecorations.Icon and InputWithDecorations.Button, use borderless versions of other components.
  *
- * @usage
+ * @example
  * <InputWithDecorations>
  *    <InputWithDecorations.Input />
  *    <InputWithDecorations.Icon>
```
Comment (mayank99) [inline packages/itwinui-react/src/core/InputWithDecorations/InputWithDecorations.tsx:87]: Can we also fix `InputWithDecorations.Icon` (nonexistent) to be `Icon`?

Comment (rohan-kadkol) [reply packages/itwinui-react/src/core/InputWithDecorations/InputWithDecorations.tsx]: Fixed (https://github.com/iTwin/iTwinUI/pull/1876/commits/314fe8af5de7f823978dd1aa1e9146b1d4afc266)

Comment (mayank99) [reply packages/itwinui-react/src/core/InputWithDecorations/InputWithDecorations.tsx]: The text in the paragraph above also includes a reference to it

Comment (rohan-kadkol) [reply packages/itwinui-react/src/core/InputWithDecorations/InputWithDecorations.tsx]: Removed that reference too (https://github.com/iTwin/iTwinUI/pull/1876/commits/2ae8f1ffc899f71fc3a03ab37756b4554a1ffdf9). Did a global search to confirm that there are no more occurrences.

File: packages/itwinui-react/src/core/InputWithDecorations/InputWithDecorations.tsx [hunk 2]
```diff
@@ -79,14 +79,14 @@ const InputWithDecorationsButton = React.forwardRef((props, ref) => {
  * Input component with various additional decorations.
  * You can add icons, buttons and other various subcomponents to it.
  *
- * If you are not using default InputWithDecorations.Icon and InputWithDecorations.Button, use borderless versions of other components.
+ * If you are not using default `Icon` and {@link InputWithDecorations.Button}, use borderless versions of other components.
```
Comment (mayank99) [inline packages/itwinui-react/src/core/InputWithDecorations/InputWithDecorations.tsx:82]: nit: These `@link` things do not show up in docs, so it's better to instead use markdown syntax (or plain text). I think it's also not usable by developers who are not working within the iTwinUI codebase.

image

**Review by mayank99 — APPROVED** [review]

File: packages/itwinui-react/src/core/InputWithDecorations/InputWithDecorations.tsx (hunk 2 above)
Comment (rohan-kadkol) [reply packages/itwinui-react/src/core/InputWithDecorations/InputWithDecorations.tsx]: Sounds good, removed (https://github.com/iTwin/iTwinUI/pull/1876/commits/70d411e741a6b2c8153c2ef09f5a262e0b582a92).

**Pull request merged**